service/src/test/java/bio/terra/cbas/controllers/TestRunSetsApiController.java

```diff
@@ -1100,7 +1096,7 @@ void oneFailedOneSucceededRun() throws Exception {
 
   @Test
   void returnErrorForUserWithNoReadAccess() throws Exception {
-    when(samService.hasReadPermission()).thenReturn(false);
+    doReturn(false).when(samService).hasReadPermission();
```

Comment:
I'm not 100% sure why, but these mocks were not overriding earlier return values when using `when`. `doReturn` resolved the problem.

only these changes seem to be necessary to make the tests pass, but I thought it was worth removing the upfront setup anyway for clarity

Reply:
I've ran into this funkiness before, from my experience it seems like in some cases `thenReturn` will call the actual method when creating the mock/spy whereas the `doReturn` doesn't have that behavior